optimum/amd/brevitas/accelerate_utils.py

```diff
@@ -414,10 +418,14 @@ def offload_params(module):
         if module._hf_hook.offload is False:
             return
         update_internal_dict(module)
-        module._hf_hook.post_forward(module, torch.tensor([]))
         for m in module.modules():
             if hasattr(m, "_hf_hook"):
-                m._hf_hook.post_forward(m, torch.tensor([]))
+                # If a module has already been offloaded (because its forward pass was called),
+                # this will raise TypeError because of tied_pointers_to_remove is None
+                try:
```

Comment:
Not a fan of this solution.
Another option would be to check whether `._hf_hook.tied_pointers_to_remove == None` as a proxy to whether `post_forward` was called.

Reply:
I am not sure to understand the issue. Is this a bug in accelerate? `tied_pointers_to_remove` is set to (at least) `set()` in the `pre_forward` so it should not be None in the post_forward.

Isn't the issue that you removed the `pre_forward` call?

Reply:
tied_pointers_to_remove is set to None in the post_forward, [here](https://github.com/huggingface/accelerate/blob/b3d2111708a480ddae60c6e1fb80f360326a0dc8/src/accelerate/hooks.py#L377).

So if you call post_forward twice (for any reasons), the second time it will raise TypeError

Reply:
Why would post_forward be called twice?

Reply:
Okay, can we remove this try/catch as this is fixed by not calling post_forward twice?

Reply:
Unfortunately post_forward might be called twice in this flow:

```
some func:
    call allocate_params:
          call pre_forward module
              call pre_forward submodule
      
     call forward on submodule (that triggers pre_forward, _old_forward and post_forward) <- weights are moved on 
         meta device

     call offload_params:
              update_dict on module and submodule's parameters <- need to check if weights are already on meta device
              call post_forward on module
              call post_forward on submodule <- here it would break as well for the reason mentioned above 
```
This happens for example when doing GPTQ

Reply:
@Giuseppe5 let's have https://github.com/huggingface/accelerate/pull/2439 merged and remove this bit.